test/e2e/httpproxy/direct_response_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e
+// +build e2e
+
+package httpproxy
+
+import (
+	contour_api_v1 "github.com/projectcontour/contour/apis/projectcontour/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/projectcontour/contour/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+func testDirectResponseRule(namespace string) {
+	Specify("direct response can be specified on route rule", func() {
+		t := f.T()
+		proxy := getDirectResponseHTTPProxy(namespace)
+		doDirectTest(namespace, proxy, t)
+	})
+}
+
+func doDirectTest(namespace string, proxy *contour_api_v1.HTTPProxy, t GinkgoTInterface) {
+	f.Fixtures.Echo.Deploy(namespace, "echo")
+
+	f.CreateHTTPProxyAndWaitFor(proxy, e2e.HTTPProxyValid)
```

Comment:
I know we haven't done this consistently elsewhere, but let's do it right here:
```suggestion
	_, ok := f.CreateHTTPProxyAndWaitFor(proxy, e2e.HTTPProxyValid)
	require.True(f.T(), ok)
```